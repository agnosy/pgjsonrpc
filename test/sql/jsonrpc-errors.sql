\set ECHO none
BEGIN;
\i sql/pgjsonrpc.sql
\set ECHO all

-- You should write your tests

SELECT jsonrpc.execute('{"jsonrpc": "2.0", "method": "foobar", "id": "1"}');
SELECT jsonrpc.execute('{"jsonrpc": "2.0", "method": "foobar, "params": "bar", "baz]');
SELECT jsonrpc.execute('{"jsonrpc": "2.0", "method": 1, "params": "bar"}');
SELECT jsonrpc.execute('
[
  {"jsonrpc": "2.0", "method": "sum", "params": [1,2,4], "id": "1"},
  {"jsonrpc": "2.0", "method"
]
');
SELECT jsonrpc.execute('[]');

ROLLBACK;
