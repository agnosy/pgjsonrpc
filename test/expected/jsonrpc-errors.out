\set ECHO none
-- You should write your tests
SELECT jsonrpc.execute('{"jsonrpc": "2.0", "method": "foobar", "id": "1"}');
                                           execute                                            
----------------------------------------------------------------------------------------------
 {"id" : "1", "jsonrpc" : "2.0", "error" : {"code" : -32601, "message" : "Method not found"}}
(1 row)

SELECT jsonrpc.execute('{"jsonrpc": "2.0", "method": "foobar, "params": "bar", "baz]');
                                         execute                                          
------------------------------------------------------------------------------------------
 {"id" : null, "jsonrpc" : "2.0", "error" : {"code" : -32700, "message" : "Parse error"}}
(1 row)

SELECT jsonrpc.execute('{"jsonrpc": "2.0", "method": 1, "params": "bar"}');
                                           execute                                            
----------------------------------------------------------------------------------------------
 {"id" : null, "jsonrpc" : "2.0", "error" : {"code" : -32600, "message" : "Invalid Request"}}
(1 row)

SELECT jsonrpc.execute('
[
  {"jsonrpc": "2.0", "method": "sum", "params": [1,2,4], "id": "1"},
  {"jsonrpc": "2.0", "method"
]
');
                                         execute                                          
------------------------------------------------------------------------------------------
 {"id" : null, "jsonrpc" : "2.0", "error" : {"code" : -32700, "message" : "Parse error"}}
(1 row)

SELECT jsonrpc.execute('[]');
                                           execute                                            
----------------------------------------------------------------------------------------------
 {"id" : null, "jsonrpc" : "2.0", "error" : {"code" : -32600, "message" : "Invalid Request"}}
(1 row)

ROLLBACK;
